### ISSUE TITLE
Found more than one resource of type 'class java.lang.class'

### ISSUE THREAD

**Issue by Loloweb** [spring-projects-experimental/spring-boot-migrator#782]
**Describe the bug**
Program returns exception "Found more than one resource of type 'class java.lang.Class. Use class org.springframework.sbm.project.resource.filter.GenericTypeListFilter instead" after finishing scan, and fails to execute any script citing previous exception.

**SBM Version**
Release v0.13.0

**SBM Application**
spring-boot-migrator.jar with VM args sbm.gitSupportEnabled=false

**Screenshots**
![image](https://user-images.githubusercontent.com/22549573/235909480-d7872a21-e90e-4a24-b18e-b2cd441a416b.png)

**Stacktrace**
```
org.springframework.sbm.project.resource.filter.ResourceFilterException: Found more than one resource of type 'class java.lang.Class'. Use class org.springframework.sbm.project.resource.filter.GenericTypeListFilter instead.
	at org.springframework.sbm.project.resource.GenericTypeFilter.apply(GenericTypeFilter.java:43)
	at org.springframework.sbm.project.resource.GenericTypeFilter.apply(GenericTypeFilter.java:27)
	at org.springframework.sbm.engine.context.ProjectContext.search(ProjectContext.java:77)
	at org.springframework.sbm.jee.jpa.actions.MigratePersistenceXmlToApplicationPropertiesAction.isApplicable(MigratePersistenceXmlToApplicationPropertiesAction.java:70)
	at org.springframework.sbm.engine.recipe.Recipe.lambda$isApplicable$0(Recipe.java:92)
	at java.base/java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1602)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230)
	at java.base/java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632)
	at org.springframework.sbm.engine.recipe.Recipe.isApplicable(Recipe.java:92)
	at org.springframework.sbm.engine.recipe.Recipes.lambda$getApplicable$2(Recipes.java:50)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:178)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
	at org.springframework.sbm.engine.recipe.Recipes.getApplicable(Recipes.java:51)
	at org.springframework.sbm.engine.commands.ApplicableRecipeListCommand.getApplicableRecipes(ApplicableRecipeListCommand.java:56)
	at org.springframework.sbm.engine.commands.ApplicableRecipeListCommand.execute(ApplicableRecipeListCommand.java:60)
	at org.springframework.sbm.shell.ScanShellCommand.scan(ScanShellCommand.java:72)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.shell.command.invocation.InvocableShellMethod.doInvoke(InvocableShellMethod.java:306)
	at org.springframework.shell.command.invocation.InvocableShellMethod.invoke(InvocableShellMethod.java:232)
	at org.springframework.shell.command.CommandExecution$DefaultCommandExecution.evaluate(CommandExecution.java:158)
	at org.springframework.shell.Shell.evaluate(Shell.java:208)
	at org.springframework.shell.Shell.run(Shell.java:140)
	at org.springframework.shell.jline.InteractiveShellRunner.run(InteractiveShellRunner.java:73)
	at org.springframework.shell.DefaultShellApplicationRunner.run(DefaultShellApplicationRunner.java:65)
	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:762)
	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:752)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1306)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1295)
	at org.springframework.sbm.SpringShellApplication.main(SpringShellApplication.java:27)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:49)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:108)
	at org.springframework.boot.loader.Launcher.launch(Launcher.java:58)
	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:65)
```

**Example Source code to replicate the issue (Optional)**
Unsure where the issue lies, so I'm unable to provide example source code.

**Desktop:**
 - OS: Ubuntu MATE (Virtual Machine)
 - Version: 22.04.2

**Additional context**
Scanned project is big and takes a good 5 minutes to scan

**Comment by BartoszMiller** [comment]
@Loloweb I faced the same issue. Deleting `persistence.xml` file from the test resouces `src/test/resources` solved the issue.

**Comment by fabapp2** [comment]
Hi @Loloweb & @BartoszMiller 
Sorry for my delayed response and thank you for reporting the issue **and** a solution to fix 🚀 
It's fixed in #966
